e2e/framework/machineset.go

```diff
@@ -191,30 +183,27 @@ func GetMachineSet(cl client.Client, name string, namespace string) (*clusterv1.
 
 	Eventually(komega.Get(machineSet), time.Minute, RetryShort).Should(Succeed(), "Failed to get machineset %s/%s.", machineSet.Namespace, machineSet.Name)
 
-	return machineSet, nil
+	return machineSet
 }
 
 // GetMachinesFromMachineSet returns an array of machines owned by a given machineSet
-func GetMachinesFromMachineSet(cl client.Client, machineSet *clusterv1.MachineSet) ([]*clusterv1.Machine, error) {
-	machines, err := GetMachines(cl)
-	if err != nil {
-		return nil, fmt.Errorf("error getting machines: %w", err)
-	}
+func GetMachinesFromMachineSet(cl client.Client, machineSet *clusterv1.MachineSet) []*clusterv1.Machine {
+	machines := GetMachines(cl)
+
 	var machinesForSet []*clusterv1.Machine
 	for key := range machines {
 		if metav1.IsControlledBy(machines[key], machineSet) {
 			machinesForSet = append(machinesForSet, machines[key])
 		}
 	}
-	return machinesForSet, nil
+
+	return machinesForSet
 }
 
 // GetNewestMachineFromMachineSet returns the new created machine by a given machineSet.
-func GetNewestMachineFromMachineSet(cl client.Client, machineSet *clusterv1.MachineSet) (*clusterv1.Machine, error) {
-	machines, err := GetMachinesFromMachineSet(cl, machineSet)
-	if err != nil {
-		return nil, fmt.Errorf("error getting machines: %w", err)
-	}
+func GetNewestMachineFromMachineSet(cl client.Client, machineSet *clusterv1.MachineSet) *clusterv1.Machine {
+	machines := GetMachinesFromMachineSet(cl, machineSet)
+    Expect(machines).ToNot(BeEmpty(), "no machines found for MachineSet %s/%s", machineSet.Namespace, machineSet.Name)
```

Comment:
Nice job adding helper functions! I see you're already following @damdo's suggestion to remove the error returns. 

I noticed a small formatting issue - the indentation looks a bit off here:

```suggestion
    Expect(machines).ToNot(BeEmpty(), "no machines found for MachineSet %s/%s", machineSet.Namespace, machineSet.Name)
```

Just moving this line to the correct indentation level to match the rest of the function! ✨